artifactory/commands/transferfiles/manager.go

```diff
@@ -182,19 +190,39 @@ func (ptm *PollingTasksManager) stop() {
 }
 
 type producerConsumerWrapper struct {
-	chunkBuilderProducerConsumer  parallel.Runner
 	chunkUploaderProducerConsumer parallel.Runner
+	chunkBuilderProducerConsumer  parallel.Runner
+	uploaderFinishedNotifier      chan bool
+	builderFinishedNotifier       chan bool
 	errorsQueue                   *clientUtils.ErrorsQueue
 }
 
+// Notifies that the uploader finished if there are no builders and the current thread is the last one
+func (pcw *producerConsumerWrapper) notifyIfUploaderFinished() {
+	if pcw.chunkBuilderProducerConsumer.ActiveThreads() == 0 && pcw.chunkUploaderProducerConsumer.ActiveThreads() == 1 {
+		pcw.uploaderFinishedNotifier <- true
+	}
+}
+
+// Notifies that the builder finished if the current thread is the last one
```

Comment:
Notifies that the builder finished if the current thread is the last one
-->
Sends a signal through the builderFinishedNotifier channel, if the chunkBuilderProducerConsumer finished running all of its tasks.

Reply:
done